### PR TITLE
DropDownEditors width behavior refactoring

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/ui/date_box/ui.date_box.strategy.native.js [hunk 1]
```diff
@@ -4,12 +4,15 @@ const support = require('../../core/utils/support');
...
 const dateSerialization = require('../../core/utils/date_serialization');
+const extend = require('../../core/utils/extend').extend;
 
 const NativeStrategy = DateBoxStrategy.inherit({
 
     NAME: 'Native',
 
-    popupConfig: noop,
+    popupConfig: function(popupConfig) {
+        return extend(popupConfig, { width: 'auto' });
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.strategy.native.js]:
```suggestion
        return extend({}, popupConfig, { width: 'auto' });
```

File: js/ui/date_box/ui.date_box.base.js [hunk 2]
```diff
@@ -263,7 +259,13 @@ const DateBox = DropDownEditor.inherit({
...
         this.callBase();
         this.$element().toggleClass(DX_AUTO_WIDTH_CLASS, !this.option('width'));
-        this._strategy?._dimensionChanged();
+        this._dimensionChanged();
+    },
+
+    _dimensionChanged: function() {
+        this.callBase(arguments);
+
+        this._popup && this._strategy._updatePopupHeight?.();
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.base.js]:
```suggestion
        if(this._popup) {
            this._strategy._updatePopupHeight?.();
        }
```

**Review by DokaRus — APPROVED** [review]

File: js/ui/drop_down_button.js [hunk 3]
```diff
@@ -411,7 +411,7 @@ const DropDownButton = Widget.inherit({
     _popupOptionChanged: function(args) {
         const options = Widget.getOptionsFromContainer(args);
 
-        this._setPopupOption('width', () => getElementWidth(this.$element()));
+        this._setPopupOption(options);
```
Comment (San4es) [inline js/ui/drop_down_button.js:414]: NOTE: consistent with https://github.com/DevExpress/DevExtreme/blob/831e3945286f6cebe4edb38c738d7423fae7f33f/js/ui/drop_down_editor/ui.drop_down_editor.js#L708

**Review by San4es — APPROVED** [review]
LGTM

**Pull request merged**